### PR TITLE
[SDF-97] Uniform Buffer Object

### PR DESCRIPTION
**Pull request by kubazuch**

### CODE REVIEW

File: libresin/libresin/core/uniform_buffer.cpp [hunk 1]
```diff
@@ -0,0 +1,44 @@
+#include <glad/gl.h>
+
+#include <libresin/core/uniform_buffer.hpp>
+
+#include "libresin/core/sdf_tree/sdf_tree.hpp"
```
Comment (migoox) [inline libresin/libresin/core/uniform_buffer.cpp:5]: Use `<...>`

**Review by migoox — APPROVED** [review]

**Comment by kubazuch** [comment]
periodt! 🙂‍↕️

File: resin/assets/test.frag [hunk 2]
```diff
@@ -16,8 +13,18 @@ uniform float u_nearPlane;
 uniform float u_farPlane;
 uniform float u_camSize;
 
+#include "blinn_phong.glsl"
+#include "sdf.glsl"
+#external_definition SDF_CODE
+
 // rendering
 const vec3 u_Ambient = vec3(0.25,0.25,0.25);
+const int kMaxNodeCount = 10;
```
Comment (migoox) [inline resin/assets/test.frag:22]: Could it be injected with `#external_definition`? If so the nodes registry would be the only place when we define the max node count

Comment (kubazuch) [reply resin/assets/test.frag:22]: Good catch, completely forgot about his variable x

**Comment by kubazuch** [comment]
I found a bug regarding transform reusing. I'm investigating more.

**Review by migoox — APPROVED** [review]
🚀🚀

**Pull request merged**